src/main/java/com/mraof/minestuck/entity/AnimatedPathfinderMob.java

```diff
@@ -0,0 +1,124 @@
+package com.mraof.minestuck.entity;
+
+import com.mraof.minestuck.entity.animation.MobAnimation;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.syncher.EntityDataAccessor;
+import net.minecraft.network.syncher.EntityDataSerializers;
+import net.minecraft.network.syncher.SynchedEntityData;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.PathfinderMob;
+import net.minecraft.world.level.Level;
+
+/**
+ * Abstract class for handling the actions of Geckolib animated mobs. It keeps track of which animation is in use via a MobAnimation and EntityDataAccessor
+ */
+public abstract class AnimatedPathfinderMob extends PathfinderMob
+{
+	/**
+	 * Retains the action from MobAnimation
+	 */
+	private static final EntityDataAccessor<Integer> CURRENT_ACTION = SynchedEntityData.defineId(AnimatedPathfinderMob.class, EntityDataSerializers.INT);
+	
+	private MobAnimation mobAnimation = MobAnimation.DEFAULT_IDLE_ANIMATION;
```

Comment:
How is this value handled across sides? Is it used or updated client-side? Is it used or updated server-side? Both? Is it being synced?

Reply:
Im having trouble remembering whether code in Goals is called on both sides or not. It is called most frequently inside of goals!

Reply:
Goals only exist server-side.

Reply:
Then CURRENT_ACTION is being updated primarily server-side, and is queried client-side in the PlayState

Reply:
Whatever my original reasoning, it doesnt appear relevant anymore! Was able to remove it without causing issues

Reply:
Alright, please commit and push that then